src/main/java/com/pubnub/api/workers/SubscribeMessageWorker.java

```diff
@@ -68,6 +68,8 @@ private void takeMessage() {
             } catch (InterruptedException e) {
                 Thread.currentThread().interrupt();
                 log.trace("take message interrupted", e);
+            } catch (Exception e) { // don't crash the thread on malformed messages
```

Comment:
I am just thinking ... maybe it would be beneficial for user to get more meaning-full error about situation when pubnub is configured with crypto and message is unencrypted similar to what we did with `history` and `fetchMessages`. In such case we could modify SubscribeMessageProcessor#processMessage method in similar fashion as we modified processMessage method in `history` and `fetchMessages`.

What do you think?  Would it bring any value?

Reply:
It sounds good if it means that the object will keep source data and a flag indicating that there were decryption errors :)

Reply:
do you mean we should actually deliver the (source) message in `listener->onMessage`, just with an error flag?

I thought the decision was to not do that for real-time messages, just history. I can do it of course if needed

Reply:
Keith said on planning that he doesn't wont to handle such situation. This is something that client should handle.
As we agreed. Let's talk about this case tomorrow on planning.

Reply:
Changed according to conversation - delivering messages with error flag.